src/objects/zcl_abapgit_object_pdts.clas.abap

```diff
@@ -0,0 +1,481 @@
+
+    ls_hrsobject-otype = c_object_type_task.
+    ls_hrsobject-objid = mv_objid.
+    INSERT hrsobject FROM ls_hrsobject.
+
+    lo_inst->save_standard_task(
+      EXPORTING
+        development_class          = iv_package
+        iv_force_gen               = abap_true
+      EXCEPTIONS
+        no_changes_allowed         = 1
+        no_client_indep_maint      = 2
+        update_error               = 3
+        insert_error_new_ts        = 4
+        new_ts_could_not_be_locked = 5
+        save_abort_by_user         = 6
+        OTHERS                     = 7 ).
+
+    check_subrc_for( `SAVE_STANDARD_TASK` ).                "#EC NOTEXT
+
+    li_document = io_xml->get_raw( ).
+
+    li_container_element = li_document->find_from_name_ns( 'CONTAINER' ).
+
+    IF li_container_element IS BOUND.
+
+      li_document = cl_ixml=>create( )->create_document( ).
+
+      li_stream = cl_ixml=>create( )->create_stream_factory( )->create_ostream_xstring( lv_xml_string ).
+
+      li_document->append_child( li_container_element ).
+
+      cl_ixml=>create( )->create_renderer(
+          document = li_document
+          ostream  = li_stream
+      )->render( ).
+
+      lo_inst->container->import_from_xml(
+        EXPORTING
+          xml_stream     = lv_xml_string
+        IMPORTING
+          exception_list = lt_exception_list ).
+
+    ENDIF.
+
+    tadir_insert( iv_package ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    CALL FUNCTION 'RH_HRSOBJECT_DELETE'
+      EXPORTING
+        act_otype           = c_object_type_task
+        act_objid           = mv_objid
+        no_confirmation_msg = abap_true
+      EXCEPTIONS
+        enqueue_failed      = 1
+        object_not_deleted  = 2
+        object_not_found    = 3
+        OTHERS              = 4.  "#EC SUBRC_OK
+
+    check_subrc_for( `RH_HRSOBJECT_DELETE` ).
+
+    "Todo: delete this comment when stable
+    "Previous code used FUNCTION 'RH_TASK_DELETE'
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~exists.
+
+    DATA: lv_endda TYPE plog-endda.
+
+    lv_endda = '99991231'.
+
+    CALL FUNCTION 'RH_READ_OBJECT'
+      EXPORTING
+        plvar     = '01'
+        otype     = c_object_type_task
+        objid     = mv_objid
+        istat     = '1'
+        begda     = sy-datum
+        endda     = lv_endda
+        ointerval = 'X'
+        read_db   = 'X'
+      EXCEPTIONS
+        not_found = 1
+        OTHERS    = 2.
+
+    rv_bool = boolc( sy-subrc = 0 ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~is_locked.
+    rv_is_locked = exists_a_lock_entry_for( iv_lock_object = 'HRSOBJECT'
+                                            iv_argument = c_object_type_task && mv_objid ).
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~is_active.
+    rv_active = abap_true.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~changed_by.
+
+    SELECT SINGLE uname
+      FROM hrs1201
+      WHERE otype = @c_object_type_task AND
+            objid = @ms_item-obj_name
+      INTO @rv_user.
```

Comment:
following suggestion will work in 702,
```suggestion
    SELECT SINGLE uname
      FROM hrs1201
      WHERE otype = c_object_type_task AND
            objid = ms_item-obj_name
      INTO rv_user.
```

Reply:
Hmm, that change errors on my 752 system:
`If the new Open SQL syntax is used, it must be used throughout. This includes using @ to escape host variables.`
Solution: INTO must come before FROM clause.